pylabrobot/resources/volume_functions.py

```diff
@@ -0,0 +1,151 @@
+# General use functions for calculating the volume of different container geometries
+
+import math
+
+def calculate_liquid_volume_container_2segments_square_vbottom(
+        x: float,
+        y: float,
+        z_pyramid: float,
+        z_cube: float,
+        liquid_height: float
+        ) -> float:
+  """
+  Calculate the volume of liquid in a container consisting of an upside-down
+  square pyramid at the bottom and a cuboid on top. The container has the 
+  same x and y dimensions for both the pyramid and the cuboid.
+     
+  Parameters:
+  x (float): The base length of the square pyramid and cube in mm.
+  y (float): The base width of the square pyramid and cube in mm.
+  z_pyramid (float): The height of the square pyramid in mm.
+  z_cube (float): The height of the cube in mm.
+  liquid_height (float): The height of the liquid in the container in mm.
+
+  Returns:
+  float: The volume of the liquid in cubic millimeters.
+
+  The function calculates the volume based on whether the liquid height is within
+  the pyramid or extends into the cuboid.
+  """
+  if liquid_height > z_pyramid+z_cube:
+    print("""WARNING: Liquid overflow detected;
+    check your labware definiton and/or that you are using the right labware.""")
+    return 0.0
```

Comment:
Should this raise an error?

Reply:
Initially it did but the automated type hinting testing flagged it as an error because the function hints at a float as the return.
So I could either remove the type hinting, or write a warning message with a float return.
I liked the later more as I find type hinting more useful, and the chance that the `liquid_height` is actually larger than the container it is in is very small (only possible if the user by accident takes the wrong labware) and the consequences of that - in this case here - are not very dangerous to the machine.

Reply:
With a `raise`, you shouldn't need a `return`. I'm pretty sure the hinting is ok with that.